### PR TITLE
Waf active rules

### PR DESCRIPTION
**Pull request by oscarDovao**
This PR contains the waf active rules implementation on the provider. Active rules are implemented using TypeSet Terraform schema type and rules can be added and deleted in bulk so only 1 or 2  requests per single update is performed.
Batch updates have not been implemented. it depends how we deal with large amount of rules, we may need batch updates.

### CODE REVIEW

File: fastly/resource_fastly_service_waf_configuration_v1.go [hunk 1]
```diff
@@ -174,6 +174,30 @@ func resourceServiceWAFConfigurationV1() *schema.Resource {
 				Description:  "XSS attack threshold.",
 				ValidateFunc: validation.IntAtLeast(1),
 			},
+			"rule": {
```
Comment (matpimenta) [inline fastly/resource_fastly_service_waf_configuration_v1.go]: I was wondering if we should make this a separate `block_` file like we did for the service. We can probably have a quick discussion on this

File: fastly/resource_fastly_service_waf_configuration_v1.go [hunk 2]
```diff
@@ -226,8 +256,28 @@ func resourceServiceWAFConfigurationV1Read(d *schema.ResourceData, meta interfac
...
+	if err != nil {
+		return err
+	}
+
+	rules := flattenWAFRules(resp.Items)
+	if err := d.Set("rules", rules); err != nil {
+		log.Printf("[WARN] Error setting WAF rules for (%s): %s", d.Id(), err)
+	}
+
+	if err = refreshWAFConfig(d, latestVersion); err != nil {
```
Comment (matpimenta) [inline fastly/resource_fastly_service_waf_configuration_v1.go]: Why do you we do this twice?

Comment (oscarDovao) [reply fastly/resource_fastly_service_waf_configuration_v1.go]: It looks like a conflict merge error. removed

File: fastly/resource_fastly_service_waf_configuration_v1.go (hunk 1 above)
Comment (oscarDovao) [reply fastly/resource_fastly_service_waf_configuration_v1.go]: I considered it, but I thought the addition of the rules wasn't adding much to the size of the file. Let's discuss...

Comment (oscarDovao) [reply fastly/resource_fastly_service_waf_configuration_v1.go]: This is changed as discussed

**Pull request closed without merging**